lib/Parse/ParsePattern.cpp

```diff
@@ -786,6 +786,10 @@ Parser::parseFunctionSignature(Identifier SimpleName,
     throwsLoc = consumeToken();
     diagnose(throwsLoc, diag::throw_in_function_type)
       .fixItReplace(throwsLoc, "throws");
+  } else if (Tok.is(tok::kw_try)) {
+    SourceLoc tryLoc = consumeToken();
+    diagnose(tryLoc, diag::throw_in_function_type)
+        .fixItReplace(tryLoc, "throws");
```

Comment:
I think we should pretend that the `throws` is correctly written. Does just modifying the condition of the previous block to `if (Tok.isAny(tok::kw_throw, kw_try))` work?

Reply:
yes, setting `throwsLoc` does work. I pushed 965605732e5ce44b8c55ebc5f85f1ec279636621